newIDE/app/src/ProjectManager/ProjectManagerItems.js

```diff
@@ -0,0 +1,304 @@
+// @flow
+import * as React from 'react';
+import { t } from '@lingui/macro';
+import { I18n } from '@lingui/react';
+import { type I18n as I18nType } from '@lingui/core';
+import WarningIcon from '@material-ui/icons/Warning';
+
+import { type MenuItemTemplate } from '../UI/Menu/Menu.flow';
+import { IconContainer } from '../UI/IconContainer';
+import { ListItem } from '../UI/List';
+import ThemeConsumer from '../UI/Theme/ThemeConsumer';
+import TextField, {
+  noMarginTextFieldInListItemTopOffset,
+} from '../UI/TextField';
+import { shouldValidate } from '../UI/KeyboardShortcuts/InteractionKeys';
+import { textEllispsisStyle } from '../UI/TextEllipsis';
+
+import { ExtensionStoreContext } from '../AssetStore/ExtensionStore/ExtensionStoreContext';
+import { type ExtensionShortHeader } from '../Utils/GDevelopServices/Extension';
+
+const styles = {
+  noIndentNestedList: {
+    padding: 0,
+  },
+  itemTextField: {
+    top: noMarginTextFieldInListItemTopOffset,
+  },
+};
+
+type ProjectStructureItemProps = {|
+  autoGenerateNestedIndicator?: boolean,
+  initiallyOpen?: boolean,
+  leftIcon?: React$Element<any>,
+  indentNestedItems?: boolean,
+  renderNestedItems: () => Array<React$Element<any> | null>,
+  primaryText: React.Node,
+  error?: ?Error,
+  onRefresh?: () => void,
+  open?: boolean,
+|};
+
+export const ProjectStructureItem = (props: ProjectStructureItemProps) => (
+  <ThemeConsumer>
+    {muiTheme => {
+      const {
+        error,
+        leftIcon,
+        onRefresh,
+        indentNestedItems,
+        ...otherProps
+      } = props;
+      return (
+        <ListItem
+          style={{
+            backgroundColor: muiTheme.listItem.groupBackgroundColor,
+            borderBottom: `1px solid ${muiTheme.listItem.separatorColor}`,
+          }}
+          nestedListStyle={
+            indentNestedItems ? undefined : styles.noIndentNestedList
+          }
+          {...otherProps}
```

Comment:
i'm usually not a big fan of using otherProps like that, it's convenient to reduce the size of a component, but when you're trying to understand how a prop is used (for instance `primaryText`), you can't find it easily with ctrl+f

Reply:
(Agree - but note that the existing code was already doing that - my fault) (so I give it a pass unless you have time and want to refactor this)